tasks/eva_3372/src/main/groovy/eva3372/CollectSplitsAndMerges.groovy

```diff
@@ -0,0 +1,178 @@
+package eva3372
+
+import org.slf4j.LoggerFactory
+import org.springframework.data.mongodb.core.BulkOperations
+import org.springframework.data.mongodb.core.query.Update
+import uk.ac.ebi.eva.accession.core.EVAObjectModelUtils
+import uk.ac.ebi.eva.accession.core.model.eva.SubmittedVariantEntity
+import uk.ac.ebi.eva.groovy.commons.EVADatabaseEnvironment
+import uk.ac.ebi.eva.groovy.commons.RetryableBatchingCursor
+
+import static org.springframework.data.mongodb.core.query.Criteria.where
+import static org.springframework.data.mongodb.core.query.Query.query
+import static uk.ac.ebi.eva.groovy.commons.EVADatabaseEnvironment.getDbsnpSveClass
+import static uk.ac.ebi.eva.groovy.commons.EVADatabaseEnvironment.getSveClass
+
+class CollectSplitsAndMerges {
+    static def logger = LoggerFactory.getLogger(CollectSplitsAndMerges.class)
+    String assembly
+    EVADatabaseEnvironment prodEnv
+    EVADatabaseEnvironment devEnv
+
+    CollectSplitsAndMerges() {}
+
+    CollectSplitsAndMerges(String assembly, EVADatabaseEnvironment prodEnv, EVADatabaseEnvironment devEnv) {
+        this.assembly = assembly
+        this.prodEnv = prodEnv
+        this.devEnv = devEnv
+    }
+
+    def collectAndCount = {
+        def evaAndDbsnpSveCursorsProd = [sveClass, dbsnpSveClass].collect { collectionClass ->
+            new RetryableBatchingCursor<>(
+                    where("seq").is(assembly).and("rs").exists(true),
+                    prodEnv.mongoTemplate, collectionClass)
+        }
+        def numRecordsProcessed = 0
+        evaAndDbsnpSveCursorsProd.each { cursor ->
+            cursor.each { List<SubmittedVariantEntity> sves ->
+                def remappedSves = sves.findAll { !it.remappedFrom.equals("") }
+                if (!remappedSves.isEmpty()) {
+                    gatherPossibleSplits(remappedSves)
+                    gatherPossibleMerges(remappedSves)
+                    numRecordsProcessed += remappedSves.size()
+                    logger.info("${numRecordsProcessed} SS processed so far...")
+                }
+            }
+        }
+        logger.info("Scan through SVE and dbsnpSVE collections complete")
+
+        detectPendingSplits()
+        detectPendingMerges()
+
+        // Counts
+        def numSplits = devEnv.mongoTemplate.count(query(where("_id").regex(/^${assembly}#/)), CveHashesWithRsId.class)
+        def numMerges = devEnv.mongoTemplate.count(query(where("_id").regex(/^${assembly}#/)), RsIdsWithCveHash.class)
+        logger.info("Processing for ${assembly} complete: ${numSplits} pending splits, ${numMerges} pending merges")
+    }
+
+    /**
+     * First step to detecting splits: store CVE hashes grouped by RS ID
+     */
+    def gatherPossibleSplits = { List<SubmittedVariantEntity> sves ->
```

Comment:
If I'm not completely mistaken this function create 1 write per rsid + 1 write per split candidate which we can approximate to 1 write per rsid since it should be much greater.
Is that right ?

Reply:
Yes that's right